backend/src/services/game.service.ts

```diff
@@ -97,3 +100,19 @@ export const allocatePlayerPunchlines = async (
 
   throw new ServiceError(ErrorType.playerId, "Player does not exist");
 };
+export const shuffleDiscardedSetups = async (game: Game): Promise<void> => {
+  if (game.setups.length <= 5) {
+    const discardedSetups: Setup[] = game.discardedSetups;
+    game.setups.push(...shuffle(discardedSetups));
+    game.discardedSetups.remove(...discardedSetups);
+    await game.save();
+  }
+};
+export const shuffleDiscardedPunchlines = async (game: Game): Promise<void> => {
+  if (game.punchlines.length <= MaxPlayers) {
```

Comment:
Technically this puts us in an invalid state. Consider the following scenario:
- We are in a game with the maximum of 50 players
- They have just played a `Pick 2` card, discarding 100 punchlines and each having 8 cards left in hand
- So at this point we know the cards in players hands total 400. The amount of discarded punchlines is at least 100, however at most could be 349. Conversely, the amount of punchlines in the deck are at least 51, at most 300.
- However, if the next round picked a setup that was `Draw 2 Pick 3`, we would need to give each player 4 cards or 200 total. This however, we only have at least 51 or at most 300 cards in the deck.

We therefore need to ensure that the number of punchlines in the deck is always at least 4 times the number of players 👌🏼

My preference however is to return the discards whenever the discard pile grows *too large* (say 50-150 in size) opposed to this approach which returns the discards whenever the deck gets *too small*. I think this offers a nicer gameplay as it prevents a card that was recently played from appearing near the top of the deck after being returned

Reply:
My problem around this is that it starts decreasing some of the randomness when it comes to selecting punchlines. If you have a 40 player game, then players are holding up to 400 cards, and the discard and deck combined will be max 100. This will essentially mean that the discard will need to be shuffled pretty much every single round.

In light of that, I think that max players of 40 might be too large. I also personally favour a more dynamic sized discard/deck shuffle approach, based loosely on the current number of players and the upcoming card rather than max size

Reply:
This may end up being a moot point if we only allow select ones for the MVP

Reply:
Yeah perhaps a limit of 25 would be more reasonable?